File: non_native/lib/repositories/i_categories_repository.dart
import '../models/category.dart';

abstract class ICategoriesRepository {
  Stream<List<Category>> getAllCategories();
  Future<Category?> getCategoryById(int id);

  Future<Category> addCategory(Category category);

  Future<void> deleteCategory(int id);
  Future<void> updateCategory(Category category);

  Future<void> incFlashcardCount(int categoryId);
  Future<void> decFlashcardCount(int categoryId);
}

File: Native/src/main/java/com/example/mindflip_native/repositories/ICategoriesRepository.kt
package com.example.mindflip_native.repositories

import com.example.mindflip_native.models.Category

interface ICategoriesRepository {
    fun getAllCategories(): List<Category>
    fun getCategoryById(id: Int): Category?
    fun addCategory(category: Category): Category
    fun deleteCategory(id: Int)
    fun updateCategory(category: Category)
    fun incFlashcardCount(id: Int)
    fun decFlashcardCount(id: Int)
}